src/libcore/num/mod.rs

```diff
@@ -2872,6 +2872,10 @@ pub enum FpCategory {
            reason = "stable interface is via `impl f{32,64}` in later crates",
            issue = "32110")]
 pub trait Float: Sized {
+    /// Type used by `to_bits` and `from_bits`.
+    #[stable(feature = "core_float_bits", since = "1.24.0")]
+    type Bits: ops::Add<Output = Self::Bits> + From<u8> + TryFrom<u64>;
```

Comment:
These bounds are pretty arbitrary. I guess it's okay for an internal trait, but alternatively could you move them to `RawFloat` (assuming that's the only place where they are needed).

Reply:
I made this work by duplicating the associated type in the `RawFloat` trait.